src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.dao.HumanDao;
+import core.basesyntax.dao.HumanDaoImpl;
+import core.basesyntax.factory.Factory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+
+        Field[] declaredFields = clazz.getDeclaredFields();
+        Class<BetDaoImpl> betDaoImplClass = BetDaoImpl.class;
+        Class<HumanDaoImpl> humanDaoImplClass = HumanDaoImpl.class;
+
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (betDaoImplClass.getAnnotation(Dao.class) != null
```

Comment:
```suggestion
                if (betDaoImplClass.isAnnotationPresent(Dao.class)
```
можна ще так перевіряти